### PR TITLE
switch_ty is redundant

### PR DESCRIPTION
**Pull request by RalfJung**
This field is redundant, but we cannot remove it currently as pretty-printing relies on it (and it does not have access to `mir::Body` to compute the type itself).

Cc @oli-obk @matthewjasper @jonas-schievink

### CODE REVIEW

File: src/librustc_middle/mir/mod.rs [hunk 1]
```diff
@@ -1075,6 +1075,8 @@ pub enum TerminatorKind<'tcx> {
         discr: Operand<'tcx>,
 
         /// The type of value being tested.
+        /// This is always the same as the type of `discr`.
+        /// FIXME: remove this redundant information. Currently, it is relied on by pretty-printing.
```
Comment (oli-obk) [inline src/librustc_middle/mir/mod.rs:1079]: If you want to be more aggressive about it you can `#[deprecate]` it.

Comment (RalfJung) [reply src/librustc_middle/mir/mod.rs:1079]: Not sure what that would help with?

Comment (oli-obk) [reply src/librustc_middle/mir/mod.rs:1079]: It would nudge people away from new usages of this field.

**Pull request merged**